.ci/travis/travis.sh

```diff
@@ -710,6 +710,16 @@ checkstyle-cli-run-openjdk14)
   exit $RESULT
   ;;
 
+spotbugs-and-pmd)
+  export MAVEN_OPTS='-Xmx2000m'
+  mvn -e clean test-compile pmd:check spotbugs:check
+  grep "Caused by:" target/site/pmd.html | cat > errors.log
```

Comment:
it is better to grep as I suggested by "Processing Errors".
in case of good build, there is no such section in report 
```
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  42.580 s
[INFO] Finished at: 2020-08-25T0:35:46-07:00
[INFO] ------------------------------------------------------------------------
✔ ~/java/github/romani/checkstyle [nmancus1/pmd-config-patch L|✚ 1] 
$ firefox target/site/pmd.html 
✔ ~/java/github/romani/checkstyle [nmancus1/pmd-config-patch L|✚ 1] 
$ cat target/site/pmd.html 
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!-- Generated by Apache Maven Doxia Site Renderer 1.7 at 2020-08-25 -->
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <title>PMD Results</title>
    <style type="text/css" media="all">
      @import url("./css/maven-base.css");
      @import url("./css/maven-theme.css");
      @import url("./css/site.css");
    </style>
    <link rel="stylesheet" href="./css/print.css" type="text/css" media="print" />
    <meta name="Date-Revision-yyyymmdd" content="20200825" />
    <meta http-equiv="Content-Language" content="en" />
        
        </head>
  <body class="composite">
    <div id="banner">
                      <div class="clear">
        <hr/>
      </div>
    </div>
    <div id="breadcrumbs">
          
                <div class="xleft">
        <span id="publishDate">Last Published: 2020-08-25</span>
                  &nbsp;| <span id="projectVersion">Version: 8.36-SNAPSHOT</span>
                      </div>
            <div class="xright">      
      </div>
      <div class="clear">
        <hr/>
      </div>
    </div>
    <div id="leftColumn">
      <div id="navcolumn">
           
                                      <a href="http://maven.apache.org/" title="Built by Maven" class="poweredBy">
        <img class="poweredBy" alt="Built by Maven" src="./images/logos/maven-feather.png" />
      </a>
                 
            </div>
    </div>
    <div id="bodyColumn">
      <div id="contentBox">
        <div class="section">
<h2><a name="PMD_Results"></a>PMD Results</h2>
<p>The following document contains the results of <a class="externalLink" href="https://pmd.github.io">PMD</a> 6.26.0.</p></div>
<p>PMD found no problems in your source code.</p>
      </div>
    </div>
    <div class="clear">
      <hr/>
    </div>
    <div id="footer">
      <div class="xright">
              Copyright &#169;                   2001&#x2013;2020.
          All rights reserved.    
                  </div>
      <div class="clear">
        <hr/>
      </div>
    </div>
  </body>
</html>

```

Reply:
Ok, no problem.  Are we concerned about printing any errors that do occur, or do we just want to fail the build?

Reply:
please just fail a build with message "Erros are detected in  target/site/pmd.html "

Reply:
Done.